Misc/NEWS.d/next/Core and Builtins/2024-05-11-21-44-17.gh-issue-118844.q2H_km.rst

```diff
@@ -0,0 +1 @@
+Added ``pycore_critical_section.h`` to ``jit.c`` to fix builds with ``--disable-gil`` and ``--enable-experimental-jit`` failing on MacOS
```

Comment:
Personally, I'd focus on the "what", rather than the "how" (since that's what users reading the changelog care about). And I don't think this issue only affects macOS.

Something like this:

```suggestion
Fix build failures when configuring with both ``--disable-gil`` and ``--enable-experimental-jit``.
```